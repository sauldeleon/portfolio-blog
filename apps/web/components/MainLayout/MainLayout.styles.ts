import styled from 'styled-components'

export const StyledPage = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  min-width: 300px;
  overflow-x: scroll;
`

export const StyledContent = styled.div`
  display: flex;
  min-height: calc(100vh - 115px - 300px);
  padding-top: 80px;
  align-items: center;
  flex-direction: column;
  overflow: hidden;

  ${({ theme }) => theme.media.up.md} {
    min-height: calc(100vh - 115px - 210px);
  }

  ${({ theme }) => theme.media.up.lg} {
    min-height: calc(100vh - 115px - 185px);
  }
`
