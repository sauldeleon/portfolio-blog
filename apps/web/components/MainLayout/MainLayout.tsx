'use client'

import { AppProgressBar as ProgressBar } from 'next-nprogress-bar'
import { Suspense } from 'react'

import { mainTheme } from '@sdlgr/main-theme'

import { StyledContent, StyledPage } from './MainLayout.styles'
import { Footer } from './components/Footer/Footer'
import { Header } from './components/Header/Header'

interface LayoutProps {
  children: React.ReactNode
}

export function MainLayout({ children }: LayoutProps) {
  return (
    <StyledPage>
      <Suspense fallback={null}>
        <ProgressBar
          color={mainTheme.colors.white}
          height="1px"
          options={{
            showSpinner: false,
            template:
              '<div class="bar" role="bar"><div class="peg"></div></div>',
          }}
          shallowRouting
        />
      </Suspense>
      <Header />
      <StyledContent>{children}</StyledContent>
      <Footer />
    </StyledPage>
  )
}
