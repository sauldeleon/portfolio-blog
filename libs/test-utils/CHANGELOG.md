# Changelog Test utils

## [0.2.3](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.2...test-utils-0.2.3) (2024-01-08)

## [0.2.2](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.1...test-utils-0.2.2) (2023-09-26)

## [0.2.1](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.0...test-utils-0.2.1) (2023-09-14)

# [0.2.0](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.1.0...test-utils-0.2.0) (2023-09-05)

### Bug Fixes

- reduced duplicated lines ([f51168a](https://github.com/sauldeleon/portfolio-blog/commit/f51168aefe025a5a65ffdb2dc22f7a16b863406e))
- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))

# 0.1.0 (2023-09-02)

### Bug Fixes

- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- add contact initial page ([fea862b](https://github.com/sauldeleon/portfolio-blog/commit/fea862b3507268013149050ae3f05fa517e7b3d9))
- add homepage ([60523aa](https://github.com/sauldeleon/portfolio-blog/commit/60523aa31ec088d318745ef0c757b4cbe656f0d4))
- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))
- improved a11y and styles ([873f656](https://github.com/sauldeleon/portfolio-blog/commit/873f656587f50df8e29ad526fdc68fea0b2c9784))
- toggle language button ([afd97e9](https://github.com/sauldeleon/portfolio-blog/commit/afd97e95beac0f70ed643c753e71db2ca014054b))
