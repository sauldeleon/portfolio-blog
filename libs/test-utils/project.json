{
  "name": "test-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/test-utils/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "version": {
      "executor": "@jscutlery/semver:version",
      "options": {
        "changelogHeader": "# Changelog Test utils \n",
        "baseBranch": "main",
        "commitMessageFormat": "chore(${projectName}): release: v${version}",
        "push": false
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/test-utils/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/test-utils/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    }
  }
}
