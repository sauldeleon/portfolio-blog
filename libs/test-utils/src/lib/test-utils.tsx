import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { Matcher } from '@testing-library/dom/types/matches'
import {
  RenderOptions,
  RenderResult,
  queryAllByAttribute as _queryAllByAttribute,
  queryByAttribute as _queryByAttribute,
  act,
  render,
  screen,
} from '@testing-library/react'
import { RouterContext } from 'next/dist/shared/lib/router-context'
import Router, { NextRouter } from 'next/router'
import { ReactNode, useEffect, useState } from 'react'
import { ThemeProvider } from 'styled-components'

import { LanguageContextProvider } from '@sdlgr/i18n-config'
import { mainTheme } from '@sdlgr/main-theme'

export type MatcherCustomFnParams = {
  content: string
  element: Element | null
}

/* istanbul ignore next */
export const renderWithTheme = (children: ReactNode): RenderResult => {
  return render(<ThemeProvider theme={mainTheme}>{children}</ThemeProvider>)
}

/* istanbul ignore next */
export function RenderProviders({ children }: { children: ReactNode }) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  })

  return (
    <QueryClientProvider client={queryClient}>
      <LanguageContextProvider value={{ language: 'en' }}>
        <ThemeProvider theme={mainTheme}>{children}</ThemeProvider>
      </LanguageContextProvider>
    </QueryClientProvider>
  )
}

/* istanbul ignore next */
export const renderApp = (
  children: ReactNode,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(<RenderProviders>{children}</RenderProviders>, options)
}

/**
 * Generic function to get element by attribute and value
 * @example queryByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function queryByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example queryByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function queryAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Generic function to get element by attribute and value
 * @example getByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function getByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example getByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function getAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Shortcut version for queryByAttribute, but specific for id
 * @param {HTMLElement} container
 * @param {string} id
 * @return {any}
 */
export function getById(container: HTMLElement, id: string) {
  return queryByAttribute('id')(container, id) as Element
}

/**
 * Shortcut version for queryByAttribute, but specific for class name
 * @param {HTMLElement} container
 * @param {string} className
 * @return {any}
 */
export function getByClassName(container: HTMLElement, className: string) {
  return queryByAttribute('class')(container, className) as Element
}

function matcherCustomFn(
  tag: string,
  matcher: Matcher,
  { element, content }: MatcherCustomFnParams
) {
  const isTag = element?.tagName.toUpperCase() === tag.toUpperCase()
  let matching = false
  if (isTag) {
    matching = content === matcher
    /* istanbul ignore else*/
    if (typeof matcher === 'function') {
      matching = matcher(content, element)
    } else if (matcher instanceof RegExp) {
      matching = matcher.test(content)
    }
  }
  return matching && isTag
}

/**
 * Get 1 element by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getByTagAndText(tag: string, matcher: Matcher) {
  return screen.getByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Get multiple elements by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getAllByTagAndText(tag: string, matcher: Matcher) {
  return screen.getAllByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Mock Math.random() to return a specific value
 * @param {number} value
 */
export function mockRandom(value = 0.1234) {
  const mockMath = Object.create(global.Math)
  mockMath.random = () => value
  global.Math = mockMath
}

/**
 * Hook to use in test to ensure "last" render is called
 */
export function useWithRefreshInTest() {
  const [withRefreshInTest, setWithRefreshInTest] = useState(false)

  useEffect(() => {
    // Change state to "rerender" the component
    setWithRefreshInTest(true)
  }, [])

  return withRefreshInTest
}

export { render } from '@testing-library/react'

/* istanbul ignore next */
export const awaitAllAsync = async () =>
  act(() => new Promise((resolve) => setTimeout(resolve, 0)))

/* istanbul ignore next */
export function renderWithNextRouter(
  children: ReactNode,
  options?: Partial<NextRouter>
) {
  return renderWithTheme(
    <RouterContext.Provider value={{ ...Router, ...options }}>
      {children}
    </RouterContext.Provider>
  )
}
