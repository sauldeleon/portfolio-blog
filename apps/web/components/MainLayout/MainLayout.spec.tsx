import { screen } from '@testing-library/react'
import { usePathname } from 'next/navigation'

import { renderApp } from '@sdlgr/test-utils'

import { MainLayout } from './MainLayout'

const mockPush = jest.fn()
jest.mock('next/navigation', () => ({
  ...jest.requireActual('next/navigation'),
  usePathname: jest.fn(),
  useRouter: () => ({
    push: mockPush,
  }),
}))

describe('MainLayout', () => {
  it('should render successfully', async () => {
    const { baseElement } = renderApp(<MainLayout>test</MainLayout>)
    await screen.findByText('test')
    expect(baseElement).toMatchSnapshot()
  })

  it('should redirect to a allowed language', async () => {
    ;(usePathname as jest.Mock).mockReturnValue('/de')
    renderApp(<MainLayout>test</MainLayout>, undefined, { language: 'de' })

    await screen.findByText('test')
    expect(mockPush).toHaveBeenCalledWith('/en/')
  })
})
