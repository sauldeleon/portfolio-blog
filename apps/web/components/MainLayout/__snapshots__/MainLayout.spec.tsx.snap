// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout should render successfully 1`] = `
.c0 {
  width: 100%;
  display: flex;
  flex-direction: column;
}

.c18 {
  display: flex;
  min-height: calc(100vh - 115px - 300px);
  padding-top: 80px;
  align-items: center;
  flex-direction: column;
  overflow: hidden;
}

.c26 {
  padding: 0;
  display: flex;
  align-items: center;
  gap: 5px;
  background: transparent;
  outline: none;
  box-shadow: none;
  color: white;
  padding-block: 0;
  padding-inline: 0;
  border: none;
  cursor: pointer;
  position: relative;
}

.c26::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: 0;
  left: 0;
  background-color: #FBFBFB;
}

.c26::after {
  transform: scaleX(0);
  transform-origin: bottom right;
  transition: transform 0.25s ease-out;
}

.c26:hover::after {
  transform: scaleX(1);
  transform-origin: bottom left;
}

.c9 {
  font-family: 'Roboto mono',monospace;
  font-weight: 400;
  color: #FBFBFB;
  font-size: 0.9rem;
  line-height: normal;
  white-space: pre-line;
}

.c9 ul,
.c9 ol {
  margin-left: 40px;
  margin-bottom: 10px;
}

.c9 ul {
  list-style: outside disc;
}

.c9 ol {
  list-style: decimal;
}

.c9 li {
  padding-left: 10px;
}

.c28 {
  position: relative;
  position: relative;
}

.c28::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: 0;
  left: 0;
  background-color: #FBFBFB;
}

.c28::after {
  transform: scaleX(0);
  transform-origin: bottom right;
  transition: transform 0.25s ease-out;
}

.c28:hover::after {
  transform: scaleX(1);
  transform-origin: bottom left;
}

.c28:hover,
.c28:active,
.c28:focus {
  text-decoration: none;
}

.c6 {
  position: relative;
  position: relative;
}

.c6::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: 0;
  left: 0;
  background-color: #FBFBFB;
}

.c6::after {
  transform: scaleX(0);
  transform-origin: bottom right;
  transition: transform 0.25s ease-out;
}

.c6:hover::after {
  transform: scaleX(1);
  transform-origin: bottom left;
}

.c6:hover,
.c6:active,
.c6:focus {
  text-decoration: none;
}

.c19 {
  position: relative;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 120px;
  margin-top: 50px;
}

.c20 {
  position: relative;
  width: 100%;
  height: 15px;
}

.c23 {
  position: absolute;
  top: -61px;
  left: 30px;
}

.c21 {
  height: 7px;
  width: 100%;
  position: relative;
}

.c21:before {
  content: '';
  position: absolute;
  width: 100%;
  height: 3px;
  background: linear-gradient(         to bottom,         rgba(0,0,0,0.3),         rgba(152,223,214,0.3)       );
}

.c22 {
  position: relative;
  max-width: 1440px;
  min-height: 120px;
  align-items: center;
  justify-content: unset;
  width: 100%;
  display: grid;
  padding: 30px;
}

.c24 {
  display: flex;
  align-items: center;
}

.c25 {
  display: flex;
  align-items: center;
  flex-grow: 1;
  flex-direction: row;
  width: 100%;
  gap: 30px;
  flex-wrap: wrap;
}

.c27 {
  height: 20px;
  display: flex;
  gap: 20px;
  justify-self: end;
  padding: 30px 0;
}

.c29::after {
  content: none;
}

.c1 {
  background-color: #FBFBFB;
  color: #000000;
  height: 130px;
  max-width: 1440px;
  width: 100%;
  padding: 0 30px;
  margin: 0 auto;
  display: flex;
  justify-items: center;
  align-items: end;
}

.c7 {
  display: flex;
}

.c7:hover,
.c7:focus {
  text-decoration: none;
}

.c4 {
  display: flex;
  align-items: end;
  flex-grow: 1;
  justify-content: space-around;
  height: 100%;
}

.c5 {
  width: 59px;
  display: flex;
  justify-content: center;
  height: 100%;
  padding-bottom: 25px;
}

.c5 .c8 {
  font-family: 'Roboto mono',monospace;
  font-weight: 500;
  color: #FBFBFB;
  font-size: 0.55rem;
  line-height: normal;
  color: #000000;
  writing-mode: tb-rl;
  transform: rotate(-180deg);
}

.c5:hover {
  background-color: #000000;
  cursor: pointer;
}

.c5:hover .c8 {
  color: #FBFBFB;
}

.c10 {
  width: 59px;
  display: flex;
  justify-content: center;
  height: 100%;
  padding-bottom: 25px;
}

.c10 .c8 {
  font-family: 'Roboto mono',monospace;
  font-weight: 500;
  color: #FBFBFB;
  font-size: 0.55rem;
  line-height: normal;
  color: #000000;
  writing-mode: tb-rl;
  transform: rotate(-180deg);
}

.c10:hover {
  background-color: #000000;
  cursor: pointer;
}

.c10:hover .c8 {
  color: #FBFBFB;
}

.c17 {
  margin-left: 8px;
  position: relative;
}

.c17::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: 0;
  left: 0;
  background-color: #FBFBFB;
}

.c17::after {
  transform: scaleX(0);
  transform-origin: bottom right;
  transition: transform 0.5s ease-out;
}

.c11 {
  outline: none;
  cursor: pointer;
  display: flex;
  align-items: center;
  max-height: 42px;
}

.c11::after {
  content: none;
}

.c11:hover circle.loading {
  stroke: white;
  stroke-width: 6px;
  stroke-dasharray: 110;
  stroke-dashoffset: 82;
  animation: kVEmCK 0.5s steps(4, jump-none) forwards;
  transform: rotate(-90deg);
  transform-origin: center;
}

.c11:hover .c16::after {
  transform: scaleX(1);
  transform-origin: bottom left;
}

.c13 {
  position: relative;
  width: 42px;
  height: 42px;
}

.c14 {
  position: absolute;
  transform: translate(-50%, -50%);
  top: 50%;
  left: 50%;
}

.c15 {
  position: absolute;
  transform: translate(-50%, -50%);
  top: 50%;
  left: 50%;
  display: flex;
}

.c2 {
  display: flex;
}

.c3 {
  color: #000000;
  margin-bottom: 25px;
}

.c12 {
  display: none;
}

@media (min-width: 768px) {
  .c18 {
    min-height: calc(100vh - 115px - 210px);
  }
}

@media (min-width: 1024px) {
  .c18 {
    min-height: calc(100vh - 115px - 185px);
  }
}

@media (min-width: 1024px) {
  .c22 {
    justify-content: space-between;
    grid-template-columns: 1fr 100px;
  }
}

@media (min-width: 1024px) {
  .c27 {
    justify-self: end;
    padding: 0;
  }
}

@media (min-width: 768px) {
  .c1 {
    align-items: center;
    background-color: transparent;
    color: #FBFBFB;
    height: 115px;
  }
}

@media (min-width: 1440px) {
  .c1 {
    padding: 0;
  }
}

@media (min-width: 768px) {
  .c7 {
    position: relative;
    position: relative;
  }

  .c7::after {
    content: '';
    position: absolute;
    width: 100%;
    height: 2px;
    bottom: 0;
    left: 0;
    background-color: #FBFBFB;
  }

  .c7::after {
    transform: scaleX(0);
    transform-origin: bottom right;
    transition: transform 0.25s ease-out;
  }

  .c7:hover::after {
    transform: scaleX(1);
    transform-origin: bottom left;
  }
}

@media (min-width: 768px) {
  .c4 {
    gap: 30px;
    align-items: center;
    justify-content: center;
    height: unset;
  }
}

@media (min-width: 768px) {
  .c5 {
    width: auto;
    padding-bottom: 0;
  }

  .c5 .c8 {
    font-family: 'Roboto mono',monospace;
    font-weight: 400;
    color: #FBFBFB;
    font-size: 0.9rem;
    line-height: normal;
    color: #FBFBFB;
    writing-mode: unset;
    transform: unset;
  }
}

@media (min-width: 768px) {
  .c10 {
    display: none;
    width: auto;
    padding-bottom: 0;
  }

  .c10 .c8 {
    font-family: 'Roboto mono',monospace;
    font-weight: 400;
    color: #FBFBFB;
    font-size: 0.9rem;
    line-height: normal;
    color: #FBFBFB;
    writing-mode: unset;
    transform: unset;
  }
}

@media (min-width: 768px) {
  .c3 {
    color: #FBFBFB;
    margin-bottom: 0;
  }
}

@media (min-width: 768px) {
  .c12 {
    display: flex;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <style>
        
          #nprogress {
            pointer-events: none;
          }

          #nprogress .bar {
            background: #FBFBFB;

            position: fixed;
            z-index: 1031;
            top: 0;
            left: 0;

            width: 100%;
            height: 1px;
          }

          /* Fancy blur effect */
          #nprogress .peg {
            display: block;
            position: absolute;
            right: 0px;
            width: 100px;
            height: 100%;
            box-shadow: 0 0 10px #FBFBFB, 0 0 5px #FBFBFB;
            opacity: 1.0;

            -webkit-transform: rotate(3deg) translate(0px, -4px);
                -ms-transform: rotate(3deg) translate(0px, -4px);
                    transform: rotate(3deg) translate(0px, -4px);
          }

          /* Remove these to get rid of the spinner */
          #nprogress .spinner {
            display: block;
            position: fixed;
            z-index: 1031;
            top: 15px;
            right: 15px;
          }

          #nprogress .spinner-icon {
            width: 18px;
            height: 18px;
            box-sizing: border-box;

            border: solid 2px transparent;
            border-top-color: #FBFBFB;
            border-left-color: #FBFBFB;
            border-radius: 50%;

            -webkit-animation: nprogress-spinner 400ms linear infinite;
                    animation: nprogress-spinner 400ms linear infinite;
          }

          .nprogress-custom-parent {
            overflow: hidden;
            position: relative;
          }

          .nprogress-custom-parent #nprogress .spinner,
          .nprogress-custom-parent #nprogress .bar {
            position: absolute;
          }

          @-webkit-keyframes nprogress-spinner {
            0%   { -webkit-transform: rotate(0deg); }
            100% { -webkit-transform: rotate(360deg); }
          }
          @keyframes nprogress-spinner {
            0%   { transform: rotate(0deg); }
            100% { transform: rotate(360deg); }
          }
        
      </style>
      <nav
        aria-label="Main menu"
        class="c1"
      >
        <a
          aria-label="Header Logo"
          class="c2"
          href="/en"
        >
          <svg
            class="c3"
            height="55"
            width="106"
          >
            slLogo.svg
          </svg>
        </a>
        <ul
          class="c4"
        >
          <li
            class="c5"
          >
            <a
              aria-label="Go to Experience page"
              class="c6 c7"
              href="/en/experience"
            >
              <p
                class="c8 c9"
              >
                Experience
              </p>
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              aria-label="Go to Contact page"
              class="c6 c7"
              href="/en/contact"
            >
              <p
                class="c8 c9"
              >
                Contact
              </p>
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              aria-label="Go to Portfolio page"
              class="c6 c7"
              href="/en/portfolio"
            >
              <p
                class="c8 c9"
              >
                CV/Portfolio
              </p>
            </a>
          </li>
          <li
            class="c10"
          >
            <a
              aria-label="Go to Portfolio page"
              class="c6 c7"
              href="/en/portfolio"
            >
              <p
                class="c8 c9"
              >
                Get in touch
              </p>
            </a>
          </li>
        </ul>
        <a
          aria-label="Go to Portfolio page"
          class="c6 c11 c12"
          href="/en/portfolio"
        >
          <div
            class="c13"
          >
            <svg
              class="c14"
              height="42"
              width="42"
            >
              circleButton.svg
            </svg>
            <div
              class="c15"
            >
              <svg>
                arrowRight.svg
              </svg>
            </div>
          </div>
          <p
            class="c8 c16 c9 c17"
          >
            Get in touch
          </p>
        </a>
      </nav>
      <div
        class="c18"
      >
        test
      </div>
      <footer
        class="c19"
        tabindex="-1"
      >
        <div
          class="c20"
        >
          <div
            class="c21"
          />
          <div
            class="c21"
          />
          <div
            class="c21"
          />
        </div>
        <div
          class="c22"
        >
          <svg
            class="c23"
            height="64"
            width="120"
          >
            slLogo.svg
          </svg>
          <nav
            aria-label="Site map"
            class="c24"
          >
            <ul
              class="c25"
            >
              <li>
                <a
                  aria-label="Go to Experience page"
                  class="c6 "
                  href="/en/experience"
                >
                  <p
                    class="c8 c9"
                  >
                    Experience
                  </p>
                </a>
              </li>
              <li>
                <a
                  aria-label="Go to Contact page"
                  class="c6 "
                  href="/en/contact"
                >
                  <p
                    class="c8 c9"
                  >
                    Contact
                  </p>
                </a>
              </li>
              <li>
                <a
                  aria-label="Go to Portfolio page"
                  class="c6 "
                  href="/en/portfolio"
                >
                  <p
                    class="c8 c9"
                  >
                    CV/Portfolio
                  </p>
                </a>
              </li>
              <li>
                <button
                  aria-label="Set Dark mode styled theme"
                  class="c26"
                >
                  <svg
                    color="#FBFBFB"
                    height="14"
                    width="14"
                  >
                    moon.svg
                  </svg>
                  <p
                    class="c8 c9"
                  >
                    Dark mode
                  </p>
                </button>
              </li>
              <li>
                <button
                  aria-label="Set Pain mode styled theme"
                  class="c26"
                >
                  <svg
                    color="#FBFBFB"
                    height="14"
                    width="14"
                  >
                    cactus.svg
                  </svg>
                  <p
                    class="c8 c9"
                  >
                    Pain mode
                  </p>
                </button>
              </li>
            </ul>
          </nav>
          <div
            class="c27"
          >
            <a
              aria-label="My Telegram username"
              class="c28 c29"
              data-testid="plain-anchor"
              href="https://t.me/Sdlgr"
            >
              <svg
                color="#FBFBFB"
              >
                telegram.svg
              </svg>
            </a>
            <a
              aria-label="My GitHub account"
              class="c28 c29"
              data-testid="plain-anchor"
              href="https://github.com/sauldeleon"
            >
              <svg
                color="#FBFBFB"
              >
                github.svg
              </svg>
            </a>
            <a
              aria-label="My LinkedIn profile"
              class="c28 c29"
              data-testid="plain-anchor"
              href="https://www.linkedin.com/in/sauldeleonguerrero"
            >
              <svg
                color="#FBFBFB"
              >
                linkedin.svg
              </svg>
            </a>
          </div>
        </div>
      </footer>
    </div>
  </div>
</body>
`;
